driver-core/src/main/java/com/datastax/driver/core/KeyspaceMetadata.java

```diff
@@ -43,7 +40,7 @@
 
     final Map<String, TableMetadata> tables = new ConcurrentHashMap<String, TableMetadata>();
     final Map<String, MaterializedViewMetadata> views = new ConcurrentHashMap<String, MaterializedViewMetadata>();
-    final Map<String, UserType> userTypes = new ConcurrentHashMap<String, UserType>();
+    final Map<String, UserType> userTypes = Collections.synchronizedMap(new LinkedHashMap<String, UserType>());
```

Comment:
👍 I like your solution.  The performance penalty is likely to be negligible in the grand scheme of things as typically there are not a lot of user types on a keyspace, and even if there were (thousands, say) this still wouldn't be noticeable I'm assuming.